### ISSUE TITLE
SC2004 - On old versions of ksh the syntax ((var)) is not valid 

### ISSUE THREAD

**Issue by antenore** [koalaman/shellcheck#1003]
I'm re-factoring some old korn shell scripts, in some cases I don't have time, or I just cannot, rewrite them in modern korn shell neither bash or other shell

I'm using the Korn Shell Version M-11/16/88f

#### For bugs
- Rule Id (if any, e.g. SC1000): SC2004
- My shellcheck version (`shellcheck --version` or "online"):  0.4.6 (last commit)
- [ X] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

This snippet should be valid:

```sh

#!/bin/ksh
(($VERBOSE)) && echo verbose

```

#### Here's what shellcheck currently says:
Line 2:
```sh
(($VERBOSE)) && echo verbose
  ^-- SC2004: $/${} is unnecessary on arithmetic variables.
```

#### Here's what I wanted or expected to see:

On ksh doing:

```sh
((VERBOSE)) &&  echo verbose
```

exits with the following error:

```sh
ksh: VERBOSE: 0403-009 The specified number is not valid for this command.
```

**Comment by orbea** [comment]
Just as a suggestion, I would do something like this instead.
```
log () { case "${VERBOSE:-0}" in 1) printf %s\\n "$@" ;; *) : ;; esac; }

log 'This is a verbose message.'
```
Should work in just about any shell.

**Comment by antenore** [comment]
@orbea yes, it does and I could adapt all the scripts we have (it's quite a gigantic task).

Shellcheck report erroneously a false problem on old KSHs. 
I could have any other kind of code that uses that same syntax, my point is that, I think, shellcheck should ignore code like that on old shells.

On most Unices, the old ksh is installed and used by default, so it'd be nice to have an exception for those cases.

Well, it's just my point of view..

**Comment by orbea** [comment]
Just to be clear I was in no way arguing if your issue is correct or not, just trying to help provide a preferable method in case you found it helpful. :)

**Comment by antenore** [comment]
Yes, of course, I appreciate it. ;-)
Do you think it'd make sense having an option to specify the shell version and therefore ignore automatically some exceptions? Or a more generic option to load a rules file (list of exceptions to be ignored)

**Comment by koalaman** [comment]
Given your example, it looks like ShellCheck is right.

`009` is not a valid octal number, and bash will fail too:

```
$ var='0403-009'; (( var ))
bash: ((: 0403-009: value too great for base (error token is "009")
```

However, given a valid arithmetic expression, like `var=0403-007`, it works fine. I imagine the same is true in Ksh88f.

I don't think you actually intended to subtract the two numbers. However, I do see that ksh93 will fall back to base 10 and therefore hide the fact that this is what's going on. 

I was unable to find a version of ksh88f to verify this on, so please reopen if I guessed wrong.
